view/rooms/app.js

```diff
@@ -266,7 +280,85 @@ function setupSocketIOClient(bots) {
   });
 } // setupSocketIOClient
 
+/**
+ * Active toggle was clicked so need to show modal.
+ *
+ * @param  {Object} event - Clicked on toggle event.
+ */
+function toggleConfirmationModal(event) {
+  event.preventDefault();
+
+  confirmationModal.setAttribute(
+    'style',
+    'display:block;'
+  );
+
+  confirmationText.innerText =
+    `Would you like to ${_isActive ? 'close' : 'activate'} this room?`;
+}
+
+// Closes the confirmation modal
+function closeConfirmationModal() {
+  confirmationModal.setAttribute(
+    'style',
+    'display:none;'
+  );
+}
+
+/**
+ * The room state has changed so it needs to be updated.
+ *
+ * @returns {Promise} For use in chaining.
+ */
+function roomStateChanged() {
+  closeConfirmationModal();
+  activeToggle.disabled = true;
+  _isActive = !_isActive;
+  const data = { active: _isActive };
+  u.patchPromiseWithUrl(GET_ROOM, data)
+  .then((res, err) => {
+    if (err) {
+      return console.log(err);
+    }
+
+    const message = _isActive ? 'Room Activated' : 'Room Closed';
+
+    const eventType = {
+      'type': 'RoomState',
+      'user': _user,
+      'active': _isActive,
+    };
+
+    const events = {
+      log: message,
+      context: eventType,
+      userId: _user.id,
+      roomId: parseInt(ROOM_ID, 10)
+    };
+
+    return u.postPromiseWithUrl('/v1/events', events);
```

Comment:
can you define events path as a constant on top of this file? similar to the existing room and bots routes